subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/NamedDomainObjectCollectionExtensions.kt

```diff
@@ -28,6 +28,34 @@ import kotlin.reflect.KProperty
 import kotlin.reflect.full.safeCast
 
 
+/**
+ * Returns a collection containing the objects in this collection of the given type. Equivalent to calling
+ * {@code withType(type).all(configureAction)}
+ *
+ * @param S The type of objects to find.
+ * @param configuration The action to execute for each object in the resulting collection.
+ * @return The matching objects. Returns an empty collection if there are no such objects
+ * in this collection.
+ * @see [NamedDomainObjectCollection.withType]
+ */
+inline fun <reified S : Any> NamedDomainObjectCollection<in S>.withType(noinline configuration: S.() -> Unit): NamedDomainObjectCollection<S> =
+    withType().also { it.all(configuration) }
```

Comment:
Why not just expose an `invoke` method on `NamedDomainObjectCollection` instead?